test/e2e/end_user_test.go

```diff
@@ -51,4 +85,65 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 		// Expect(kerrors.IsForbidden(err)).To(Equal(true))
 		fmt.Printf("PDB create error: %v\n", err)
 	})
+
+	It("should deploy a template and check the visit counter increments", func() {
+		var regex = regexp.MustCompile(`\<span.*id=\"count-value\"\>(?P<counter>\d+)\<\/span\>`)
+
+		// Create the template
+		template, err := c.tc.Templates("openshift").Get(
+			"nodejs-mongodb-example", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Create a secret to hold the template params
+		secret, err := c.cc.Secrets(c.namespace).Create(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "parameters",
+			},
+			StringData: map[string]string{
+				"NAME":         c.namespace,
+				"MEMORY_LIMIT": "1024Mi",
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Instantiate the template
+		_, err = c.tc.TemplateInstances(c.namespace).Create(
+			&templatev1.TemplateInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: c.namespace,
+				},
+				Spec: templatev1.TemplateInstanceSpec{
+					Template: *template,
+					Secret: &corev1.LocalObjectReference{
+						Name: secret.Name,
+					},
+				},
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Poll for the deployment status
+		err = wait.PollImmediate(2*time.Second, 2*time.Minute, c.templateInstanceIsReady)
```

Comment:
2*time.Minute is too short, you'll flake

Reply:
5 or 10 minutes should be fine to use.

Reply:
test has been taking between 90 and 100 seconds consistently, going to use 5min here